### PR TITLE
swarm/fuse: ignore default manifest entry

### PR DESCRIPTION
**Pull request by cobordism**
this fixes the problem I was having in fuse where files would show up, but directories would not.

The issue was caused by manifests with default entries (i.e. manifests in which an entry has an empty 'path'). FUSE cannot handle files without a name. This PR simply ignores the default entry in a manifest.
In future we may think of better solutions the represent the defaultpath in a filesystem.

### CODE REVIEW

File: swarm/fuse/swarmfs_unix.go [hunk 1]
```diff
@@ -120,6 +120,10 @@ func (swarmfs *SwarmFS) Mount(mhash, mountpoint string) (*MountInfo, error) {
 
 	log.Trace("swarmfs mount: traversing manifest map")
 	for suffix, entry := range manifestEntryMap {
+		if suffix == "" { //empty suffix means that the file has no name - i.e. this is the default entry in a manifest. Since we cannot have files without a name, let us ignore this entry
+			log.Warn(fmt.Sprintf("Manifest has a default entry with hash %v. This is ignored and not shown in FUSE", entry.Hash))
```
Comment (holisticode) [inline swarm/fuse/swarmfs_unix.go]: I would change the log text to:

"Manifest has an empty path entry with default entry %v, which would cause FUSE to mount improperly. Ignoring"

Comment (holisticode) [reply swarm/fuse/swarmfs_unix.go]: Then in the comment, please add a bit more descriptive information what happens if this is not done, i.e. what you described in the chat, in some short form. I did not immediately see that this change in fact actually fixes the problem :)

"actually the mounting works, but you only see a few files (random sample) and no directories and an ls command gives an input/output error... but if you do ls subdirectory/ then that works again.
in short, the behaviour is erratic"

**Review by holisticode — APPROVED** [review]

File: swarm/fuse/swarmfs_unix.go [hunk 2]
```diff
@@ -120,6 +120,10 @@ func (swarmfs *SwarmFS) Mount(mhash, mountpoint string) (*MountInfo, error) {
 
 	log.Trace("swarmfs mount: traversing manifest map")
 	for suffix, entry := range manifestEntryMap {
+		if suffix == "" { //empty suffix means that the file has no name - i.e. this is the default entry in a manifest. Since we cannot have files without a name, let us ignore this entry
+			log.Warn(fmt.Sprintf("Manifest has an empty-path (default) entry, which would case FUSE mount to fail. Ignoring entry %v", entry.Hash))
```
Comment (zelig) [inline swarm/fuse/swarmfs_unix.go]: would case -> would cause

**Review by zelig — CHANGES_REQUESTED** [review]

File: swarm/fuse/swarmfs_unix.go (hunk 2 above)
Comment (zelig) [reply swarm/fuse/swarmfs_unix.go]: * use the structured logging. no need to use `Sprintf`
* use a simpler message
* better put the context, like what hash is being mounted or under what mountpoint. the entry's hash is the least relevant

`log.Warn("Ignore manifest entry with empty path",  "mount point", mountpoint, "swarm hash", hashToMount)`

Comment (cobordism) [reply swarm/fuse/swarmfs_unix.go]: > use the structured logging.

I don't know what that means. I copied the Sprintf syntax from elsewhere in the code.

>  the entry's hash is the least relevant

how so? This hash represents the one piece of data that is missing from the fuse mount.


I'll make a simpler message with no info.

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: bce5a384ea5474c8dfd531f086dedd0d65ecead5
  * Commits must be in the following format: **%{scope}: %{description}**

Guidelines are available at https://github.com/ethersphere/go-ethereum

---
This message was auto-generated by https://gitcop.com

**Review by nonsense — APPROVED** [review]
